service/src/main/java/uk/gov/hmcts/reform/fpl/service/ChildrenService.java

```diff
@@ -30,4 +39,38 @@ public String getChildrenLabel(List<Element<Child>> children) {
     public void addPageShowToCaseDetails(CaseDetails caseDetails, List<Element<Child>> children) {
         caseDetails.getData().put("pageShow", children.size() <= 1 ? "No" : "Yes");
     }
+
+    public List<Element<Child>> updateFinalOrderIssued(List<Element<Child>> children,
+        String orderAppliesToAllChildren, ChildSelector childSelector) {
+
+        if (YES.getValue().equals(orderAppliesToAllChildren)) {
+            return children.stream()
+                .map(child -> updateFinalOrderIssuedOnChild(child, YES))
+                .collect(Collectors.toList());
+        } else {
+            List<Integer> selectedChildren = childSelector != null ? childSelector.getSelected() : new ArrayList<>();
+            return IntStream.range(0, children.size())
+                .mapToObj(index -> {
+                    Element<Child> child = children.get(index);
+                    if (!selectedChildren.isEmpty() && selectedChildren.contains(index)) {
+                        return updateFinalOrderIssuedOnChild(child, YES);
+                    } else if (StringUtils.isEmpty(child.getValue().getParty().getFinalOrderIssued())) {
+                        return updateFinalOrderIssuedOnChild(child, NO);
+                    }
+                    return child;
+                })
+                .collect(Collectors.toList());
+        }
+    }
+
+    private Element<Child> updateFinalOrderIssuedOnChild(Element<Child> child, YesNo yesNo) {
+        ChildParty updatedChildParty = child.getValue().getParty().toBuilder()
+            .finalOrderIssued(yesNo.getValue()).build();
+        Child updatedChild = child.getValue().toBuilder()
+            .party(updatedChildParty).build();
+        return Element.<Child>builder()
+            .id(child.getId())
+            .value(updatedChild)
+            .build();
+    }
```

Comment:
lets just have a setter on the child object rather than all this to builder

Reply:
then our flow can be more like 
```
if (for all)  {children.stream().map(setter for all)... }
if (yes) {find child to set yes ... set yes }
else set no
```

Reply:
I hadn't come across any use of setters on the model object. So I assumed the coding style for this project was to use builders. I will change to setter.